src/app/components/Settings/NodeSettings.tsx

```diff
@@ -0,0 +1,209 @@
+// Huge bastardization of ant design's Transfer component
+import React from 'react';
+import { connect } from 'react-redux';
+import { Form, Input, Button, Icon, Drawer, message } from 'antd';
+import { editNodeField, updateNodeUrl, updateMacaroons } from 'modules/node/actions';
+import { changePassword, setPassword, cancelChangePassword } from 'modules/crypto/actions';
+import CreatePassword from 'components/CreatePassword';
+import InputAddress from 'components/SelectNode/InputAddress';
+import UploadMacaroons from 'components/SelectNode/UploadMacaroons';
+import { AppState } from 'store/reducers';
+
+interface StateProps {
+  url: AppState['node']['url'];
+  readonlyMacaroon: AppState['node']['readonlyMacaroon'];
+  adminMacaroon: AppState['node']['adminMacaroon'];
+  isNodeChecked: AppState['node']['isNodeChecked'];
+  editingNodeField: AppState['node']['editingNodeField'];
+  isUpdatingNodeUrl: AppState['node']['isUpdatingNodeUrl'];
+  updateNodeUrlError: AppState['node']['updateNodeUrlError'];
+  isUpdatingMacaroons: AppState['node']['isUpdatingMacaroons'];
+  updateMacaroonsError: AppState['node']['updateMacaroonsError'];
+  isChangingPassword: AppState['crypto']['isChangingPassword'];
+  password: AppState['crypto']['password'];
+}
+
+interface DispatchProps {
+  editNodeField: typeof editNodeField;
+  updateNodeUrl: typeof updateNodeUrl;
+  updateMacaroons: typeof updateMacaroons;
+  changePassword: typeof changePassword;
+  cancelChangePassword: typeof cancelChangePassword;
+  setPassword: typeof setPassword;
+}
+
+type Props = StateProps & DispatchProps;
+
+class NodeSettings extends React.Component<Props> {
+
+  componentWillUpdate(nextProps: Props) {
+    const { 
+      isNodeChecked, 
+      isChangingPassword, 
+      password, 
+      isUpdatingMacaroons, 
+      adminMacaroon, 
+      readonlyMacaroon 
+    } = this.props;
+
+    if (isNodeChecked !== nextProps.isNodeChecked && nextProps.isNodeChecked) {
+          // isNodeChecked false -> true
+          message.success(`Connected to ${nextProps.url}`, 2);
+    }
+    if (isChangingPassword !== nextProps.isChangingPassword && !isChangingPassword
+        && password !== nextProps.password) {
+          // isChangingPassword true -> false and password changed
+          message.success('Password Updated', 2);
+    }
+    const macaroonsChanged = adminMacaroon !== nextProps.adminMacaroon
+      || readonlyMacaroon !== nextProps.readonlyMacaroon;
+    if (isUpdatingMacaroons && macaroonsChanged) {
+        // isUpdatingMacaroons is true and a macaroon changed
+        message.success('Macaroons Updated', 2);
+    }
+  }
```

Comment:
I don't like this 💩 code at all. I have some thoughts on how to improve it throughout the app using an alerts reducer, but I'll save that discussion for a separate issue/PR.